cg/store/crud/read.py

```diff
@@ -187,6 +194,19 @@ def get_analysis_by_case_entry_id_and_completed_at(
             completed_at_date=completed_at_date,
         ).first()
 
+    def get_latest_started_analysis_for_case(self, case_id: str) -> Analysis:
+        """Return the latest started analysis for a case.
+        Raises:
+            AnalysisDoesNotExistError if no analysis is found.
+        """
+        case: Case = self.get_case_by_internal_id(case_id)
+        analyses: list[Analysis] = case.analyses
+        if not analyses:
+            raise AnalysisDoesNotExistError(f"No analysis found for case {case_id}")
+        analyses.sort(key=lambda x: x.started_at, reverse=True)
```

Comment:
```suggestion
        started_analyses: list[Analysis] = [a for a in analyses if a.started_at is not None]
        started_analyses.sort(key=lambda a: cast(datetime, a.started_at), reverse=True)
        analysis: Analysis | None = started_analyses[0]
```
Since it's possible for started_at to not be set I think we need to handle it, otherwise this method would throw an exception when trying to sort the list. I propose that we first filter them out and then I add a `cast` to convince the type checker that we at this point are sure that there will be no null-values.

Reply:
set it as out of scope